aws/cloudformation/drone-stack.yml

```diff
@@ -0,0 +1,367 @@
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ELB Allowed Ports
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: 0.0.0.0/0
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - ALB SecurityGroup"
+
+  DroneMasterTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      Name: !Sub "TargetGroup-${AWS::StackName}"
+      VpcId: !Ref VpcId
+      Port: 80
+      Protocol: HTTP
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - TargetGroup"
+
+  DroneMasterELB:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: !Sub "ELB-${AWS::StackName}"
+      SecurityGroups:
+        - !Ref ElbSecurityGroup
+      Subnets: !Ref SubnetIds
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - ELB"
+
+  LoadBalancerListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      LoadBalancerArn: !Ref DroneMasterELB
+      Port: 80
+      Protocol: HTTP
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref DroneMasterTargetGroup
+
+# Drone Master resources:
+# Security Group / EC2 Launch Template / EC2 Autoscaling Group
+# ECS cluster / ECS service / ECS task definition
+
+  DroneMasterEcsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ECS Allowed Ports
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        -
+          IpProtocol: tcp
+          FromPort: 1
+          ToPort: 65535
+          SourceSecurityGroupId: !Ref ElbSecurityGroup
+        -
+          IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+
+  DroneMasterECSCluster:
+    Type: AWS::ECS::Cluster
+
+  DroneMasterLaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
+    Properties:
+      LaunchTemplateName: !Sub "DroneMasterLaunchTemplate-${AWS::StackName}"
+      LaunchTemplateData: 
+        IamInstanceProfile: 
+          Name: ecsInstanceRole
+        KeyName: !Sub "winter-dev-${AWS::Region}"
+        SecurityGroupIds: 
+          - !Ref DroneMasterEcsSecurityGroup
+        ImageId: ami-0c4b05c0409735671
+        InstanceType: t3.small
+        Monitoring:
+          Enabled: true
+        UserData: 
+          Fn::Base64:
+            !Sub 
+              - "#!/bin/bash\necho ECS_CLUSTER=${EcsClusterName} >> /etc/ecs/ecs.config;echo ECS_BACKEND_HOST= >> /etc/ecs/ecs.config;"
+              - {EcsClusterName: !Ref DroneMasterECSCluster}
+ 
+  DroneMasterEcsInstanceAsg:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      AvailabilityZones: !Ref AZs
+      LaunchTemplate:
+        LaunchTemplateId: !Ref DroneMasterLaunchTemplate
+        Version: !GetAtt DroneMasterLaunchTemplate.LatestVersionNumber
+      MinSize: 1
+      MaxSize: 1
```

Comment:
Is this auto-scaling group needed to ensure high-availability of the single instance, or will the count potentially be increased in the future?

Reply:
We likely won't be increasing this count, since right now it just uses a sqlite DB directly on the master instance. So more the former.

Of course because the DB is stored on the master instance, it's not actually ephemeral (unless we added something like EFS), but the manual work to configure a new master instance wouldn't be much. Given that, does it still make sense to use an ASG here? I dont know how cloudformation handles managing a single instance.